### ISSUE TITLE
BN

### ISSUE THREAD

**Issue by liu666666** [yjxiong/temporal-segment-networks#106]
Thanks for your nice work!
I have two questions:
1.The value of batch size in train.prototxt is 24, I use 4 gpus with openmpi, and the value of iter_size in solver.prototxt is 2. So my real batch_size is 24*4*2=192.Right?
2.The function of BatchNorm layer is to normalise the input to have 0-mean and/or unit (1) variance across the batch. what is the batch in your code If I have set the parameters as mentioned in (1)? 24 or 96 or 192?

**Comment by yjxiong** [comment]
Where did you get that 24 and itersize 2? All models in this repo use 32 local batchsize, which yields 128 effective batchsize with 4 GPUS.

Can't understand your second question. Please clarify.

**Comment by liu666666** [comment]
Thanks for your replay!
(1)I have to set the value of batch size in train.prototxt to 24 during GPU memory limits. with your nice code,I use 4 gpus with openmpi, and the value of iter_size in solver.prototxt is 2. So my real batch_size is 24**4**2=192.Right?
(2)The function of BN layer in your code is to normalise the input to have 0-mean and/or unit (1) variance across the batch. what is the batch in your code If I have set the parameters as mentioned in (1)?namely,batchsize is 24,4GPUs,iter_size is 2. The batch is which one?24 or 96 or 192?

**Comment by yjxiong** [comment]
(1) Yes.

(2) BN perform normalization independently on each GPU and iteration. That is, it only observes `24` . This is ok for most cases.

**Comment by liu666666** [comment]
Thanks for your kind help.
